### PR TITLE
Refactor the robot tester

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: model/robot_test_results.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	Y int
+}
+
+func (p Position) Equals(pos Position) bool {
+	return p.X == pos.X && p.Y == pos.Y
+}
+
+type TypedPosition struct {
+	Position
+	Type string
```
Comment (flowlo) [inline model/robot_test_results.go]: You have some magic strings in your code. Please make `Type` a type (`type Type string`) and define constants, just like you did for the directions (`Up`, `Right`, ...).

File: test/robot.go [hunk 2]
```diff
@@ -2,215 +2,155 @@ package test
...
+			if ok := move(&tr, n, direction, m); !ok {
+				return tr, nil
 			}
 		case "right":
 			direction = turnRight(direction)
 		case "left":
 			direction = turnLeft(direction)
 		case "pick":
-			moves = pick(moves, m.Coins)
+			picks = append(picks, len(tr.Moves)-1)
```
Comment (flowlo) [inline test/robot.go]: Here you do not append anything to `tr.Moves`, so when iterating through `tr.Maps` to check the picks later on, you'll likely override some other elements `Event` property.

Comment (victorbalan) [inline test/robot.go]: You can override only the move events. I`ll check for double picks.

Comment (flowlo) [inline test/robot.go]: But still, you override *something*, right? That's not good.

**Pull request merged**